buddyboss/buddyboss-platform
# Fixed - Forum association with groups

### All Submissions:






### Changes proposed in this Pull Request:



Fixes #2112.

### How to test the changes in this Pull Request:

**Check Video**: https://www.loom.com/share/cbb6379fa57d45e3a8369860416398d6


- Do not allow same Forum to be associated with more than 1 Group. (already scoped above)

- Do not allow to associate sub-forums or any level child forums to any group no matter if parent forum associated with Group or No.
Do not allow to associate Forums Type Category also with the Group.
Needs to be taken care of Adding/Editing Group backend, Adding/Editing Group Frontend.

- When Forum associated with Group then Forum and all level child Forums "Visibility" Field should be disabled.

- Disable "Forum Parent" and "Type" Field when the main Forum associated with Group.


- Group Forum's Sub-Forums will inherit the same privacy as Group.
Layout will be same as Parent forum in Groups.
Single Sub-Forum URL in Group:
domain.com/groups/GROUP_SLUG/forums/SUB_FORUM_SLUG
Sub-Forum Single Discussion URL in Group (same as parent forum discussion):
domain.com/groups/GROUP_SLUG/forums/topic/DISCUSSION_SLUG/

- Sub-Forums and all level child forums and discussions always should be linked to Group URL if associated same as Parent Forum from Backend and Frontend.

- Make sure everything here is backward compatible.
IF currently one forum associated with multiple Group then let it work as it is otherwise it will break the existing customer site.
	- Same set of discussion show for both the groups
	- View Forum redirects to any one of the Group. Maybe the one assigned last. That's how it is.
	- Same workflow for child forums.
IF currently any sub-forum associated with Group then also it should work fine.

- Make sure all change here works fine with Forum settings "Allow social groups to have their own forums" both enabled or disabled.
- Settings > Forums > Group Forums Parent, Remove this settings option from BB settings panel.
Make sure everything here is backward compatible.

### Proof Screenshots or Video
1. Do not allow same Forum to be associated with more than 1 Group.
https://www.loom.com/share/514dd48fbf534fe5a79f78470cc2c460

2. Do not allow to associate sub-forums or any level child forums to any group no matter if parent forum associated with Group or No.
Do not allow to associate Forums Type Category also with the Group.
https://www.loom.com/share/0a06a56453b84e0e95eb769cb6d581b1

3. When Forum associated with Group then Forum and all level child Forums "Visibility" Field should be disabled.
https://www.loom.com/share/232aaec6c53a4783a44eab43ea03dc30

4. Disable "Forum Parent" and "Type" Field when the main Forum associated with Group.
https://www.loom.com/share/915c2332012944e8bdb7fa59e63c9455

5. Group Forum's Sub-Forums will inherit the same privacy as Group.
https://www.loom.com/share/782df6ed23ae42c5b850f176df246993

6. Sub-Forums and all level child forums and discussions always should be linked to Group URL if associated same as Parent Forum from Backend and Frontend.
https://www.loom.com/share/e58906445715465fb2a83f4f731c8908

7. All change here works fine with Forum settings "Allow social groups " both enabled or disabled.
https://www.loom.com/share/754e3ac467c042eb8047f2af9ed49484



### Other information:




### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.


### Related Platform API PR
https://github.com/buddyboss/buddyboss-platform-api/pull/300

### Related Theme PR
https://github.com/buddyboss/buddyboss-theme/pull/1345

### Related App Native Issue
https://github.com/buddyboss/buddyboss-app-native/issues/1518
